MapleServer2/PacketHandlers/Game/DungeonHandler.cs

```diff
@@ -48,12 +60,80 @@ public override void Handle(GameSession session, PacketReader packet)
                     break;
             }
         }
+        public static void HandleEnterDungeonPortal(GameSession session)
+        {
+            int instanceId = session.Player.InstanceId;
+            DungeonSession dungeonSession = GameServer.DungeonManager.GetDungeonSessionByInstanceId(instanceId);
+            if (dungeonSession == null)
+            {
+                return;
+            }
+            session.Player.Warp(mapId: dungeonSession.DungeonMapIds.First(), instanceId: dungeonSession.DungeonInstanceId);
 
-        public static void HandleEnterDungeonRoom(GameSession session, PacketReader packet)
+        }
+        public static void HandleCreateDungeon(GameSession session, PacketReader packet)
         {
             int dungeonId = packet.ReadInt();
+            bool groupEnter = packet.ReadBool();
+            Player player = session.Player;
```

Comment:
Argh, git deleted my text, so again:
They are both used in if checks in the code below, e.g.
`if (player.DungeonSessionId != -1)`
` if (groupEnter)`
These checks can be removed later, because the client should prevent creating a new dungeon by greying out the fields in the dungeon directory. The greying out happens on senddungeonroom, I am convinved, but I have not decyphered those packets yet, as they are long.